packages/connect-node/src/node-universal-handler.ts

```diff
@@ -89,12 +89,15 @@ export function universalRequestFromNodeRequest(
     parsedJsonBody !== undefined
       ? parsedJsonBody
       : asyncIterableFromNodeServerRequest(nodeRequest);
+  const abortController = new AbortController();
+  nodeRequest.on("close", () => abortController.abort());
```

Comment:
We have to honor HTTP/2 error codes here. We can use `connectErrorFromH2ResetCode()` added in https://github.com/bufbuild/connect-es/pull/619.

For example:

```ts
  if ("stream" in nodeRequest) {
    // HTTP/2 has error codes we want to honor
    nodeRequest.on("close", () => {
      const err = connectErrorFromH2ResetCode(nodeRequest.stream.rstCode);
      if (err) {
        ac.abort(err);
      } else {
        ac.abort();
      }
    });
  } else {
    // HTTP/1.1 does not
    nodeRequest.on("close", () => ac.abort());
  }
```

Yes, I expect the "close" event to always fire.

Aborting with a reason is not supported in Node.js v16, but it will be EOL soon. For the interim, it will be fine if v16 does not bubble up the code correctly.

We should update `universalServerRequestFromFetch()` as well, and use the signal from the fetch Request.